packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -295,6 +325,39 @@ function currentDirectoryFilepath(filename: string) {
   return Uri.joinPath(currentDir, filename);
 }
 
+function findSelectionContent(): FoamSelectionContent | undefined {
+  const editor = window.activeTextEditor;
+  if (editor === undefined) {
+    return undefined;
+  }
+
+  const document = editor.document;
+  const selection = editor.selection;
+
+  if (!document || selection.isEmpty) {
+    return undefined;
+  }
+
+  return {
+    document,
+    selection,
+    content: document.getText(selection),
+  };
+}
+
+async function replaceSelectionWithWikiLink(
+  document: TextDocument,
+  selection: Selection
+) {
+  const newNoteFile = window.activeTextEditor.document.fileName;
```

Comment:
We already have this filepath computed (we pass it into `writeTemplate`).

Instead of relying on the behaviour of VS Code in this line, let's pass the filepath into `replaceSelectionWithWikiLink` and use that.